chainer/dataset/convert.py

```diff
@@ -139,3 +145,106 @@ def _concat_arrays_with_padding(arrays, padding):
             result[(i,) + slices] = src
 
     return result
+
+
+class ConcatWithAsyncTransfer(object):
+
+    """Interface to concatenate data and transfer them to GPU asynchronously.
+
+    It enables to transfer next batch of input data to GPU while GPU is
+    running kernels for training using current batch of input data.
+    """
```

Comment:
Write the documentation of the `stream` argument.

Reply:
will add docs about the `stream`